index.bs

```diff
@@ -535,66 +621,154 @@ To determine whether an |attribute| <dfn>contains a javascript: URL</dfn>:
 
 </div>
 
-## Configuration Processing ## {#configuration-processing}
+## Configuration Modifyers ## {#configuration-modifyers}
 
 <div algorithm>
-To <dfn for="SanitizerConfig">allow an element</dfn> |element| with a {{SanitizerConfig}} |configuration|, do:
+To <dfn for="Sanitizer">add an item</dfn> |item| to a SanitizerConfig |config| list named |list|:
+
+1. If |config|[|list|] [=map/exists=]:
+  1. If |config|[|list|] does not [=contain=] |item|:
+    1. [=list/Append=] |item| to |config|[|list|].
+    1. Return true.
+1. Return false.
 
-1. Set |element| to the result of [=canonicalize a sanitizer element with attributes=] with |element|.
-1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/elements}}"].
-1. [=list/Append=] |element| to |configuration|["{{SanitizerConfig/elements}}"].
-1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/removeElements}}"].
-1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"].
+</div>
 
-NOTE: Handling of [=allowElement=] is a little more complicated than the other
-    methods, because the element allow list can have per-element allow- and
-    remove-attribute lists. We first remove the given element from the list
-    before then adding it, which has the effect of re-setting (rather than
-    merging or elsehow modifying) the per-element list to whatever is passed
-    in. In other words, the per-element allow- and remove-lists can only be
-    set as a whole.
+<div algorithm>
+To <dfn for="Sanitizer">remove an item</dfn> |item| from a SanitizerConfig |config| list named
+|list|:
 
-NOTE: [=SanitizerConfig/Remove=] matches on name and namespace, so adding an
-    element with attributes would still remove the matching element from the
-    {{SanitizerConfig/removeElements}} and {{SanitizerConfig/replaceWithChildrenElements}} lists.
+1. If |config|[|list|] [=map/exists=]:
+  1. If |config|[|list|] [=contains=] |item|:
+    1. [=list/Remove=] |item| to |config|[|list|].
+    1. Return true.
+1. Return false.
 
 </div>
 
+
+<div algorithm>
+To <dfn for="SanitizerConfig">allow an element</dfn> |element| with a
+{{SanitizerConfig}} |configuration|, do:
+
+1. [=Assert=]: |configuration| is [=SanitizerConfig/valid=].
+1. Set |element| to the result of [=canonicalize a sanitizer element with
+    attributes=] with |element|.
+1. If |configuration|["{{SanitizerConfig/elements}}"] [=map/exists=]:
+    1. [=Add an item=] |element| to |configuration| list "{{SanitizerConfig/elements}}".
+    1. [=Remove an item=] |element| from |configuration| list
+        "{{SanitizerConfig/replaceWithChildrenElements}}"
+    1. **IF DISALLOW REDUNDANT PER-ELEMENT-ATTRIBUTES**:
+        1. If |element| [=map/contains=] {{SanitizerElementNamespaceWithAttributes/attributes}}
+            and |configuration| [=map/contains=] {{SanitizerConfig/attributes}}:
+            1. [=list/iterate|For each=] |attr| in
```

Comment:
As mentioned in the meeting: We need to iterate on the global lists and change the local lists.

Reply:
Updating of local lists has been added.